src/rules/importBlacklistRule.ts

```diff
@@ -54,9 +89,140 @@ export class Rule extends Lint.Rules.AbstractRule {
 }
 
 function walk(ctx: Lint.WalkContext<string[]>) {
+    type Options = Array<string | { [moduleName: string]: string[] }>;
+    interface BannedImports {
+        [moduleName: string]: true | Set<string>;
+    }
+
+    // Merge/normalize options.
+    // E.g., ["a", { "b": ["c"], "d": ["e"] }, "f", { "f": ["g"] }]
+    // becomes { "a": true, "b": Set(["c"]), "d": Set(["e"]), "f": true }.
+    const bannedImports = (ctx.options as Options).reduce(
+        (acc, it) => {
+            if (typeof it === "string") {
+                acc[it] = true;
+            } else {
+                Object.keys(it).forEach((moduleName) => {
+                    if (acc[moduleName] instanceof Set) {
+                        it[moduleName].forEach((bannedImport) => {
+                            (acc[moduleName] as Set<string>).add(bannedImport);
+                        });
+                    } else if (acc[moduleName] !== true) {
+                        acc[moduleName] = new Set(it[moduleName]);
+                    }
+                });
+            }
+            return acc;
+        },
+        {} as BannedImports, // tslint:disable-line no-object-literal-type-assertion
```

Comment:
`.reduce<BannedImports>(...)` should allow you to remove this disable.

Reply:
Fixed